### PR TITLE
[WINDUPRULE-440,443,444,446,447,448,450,451,452,453,454,455,456] Add rules for java generic information chapter

### PR DESCRIPTION
**Pull request by mmelko**
https://issues.redhat.com/browse/WINDUPRULE-440
https://issues.redhat.com/browse/WINDUPRULE-443
https://issues.redhat.com/browse/WINDUPRULE-444
https://issues.redhat.com/browse/WINDUPRULE-446
https://issues.redhat.com/browse/WINDUPRULE-447
https://issues.redhat.com/browse/WINDUPRULE-448
https://issues.redhat.com/browse/WINDUPRULE-450
https://issues.redhat.com/browse/WINDUPRULE-451
https://issues.redhat.com/browse/WINDUPRULE-452
https://issues.redhat.com/browse/WINDUPRULE-453
https://issues.redhat.com/browse/WINDUPRULE-454
https://issues.redhat.com/browse/WINDUPRULE-455
https://issues.redhat.com/browse/WINDUPRULE-456

### CODE REVIEW

**Comment by mmelko** [comment]
This PR depends on #401 due to https://github.com/windup/windup-rulesets/pull/409/commits/7e3c5a02c886fa2e1b6c7841175205403108239b that's adding rule also into `xml-moved-components.windup.xml`

**Comment by m-brophy** [comment]
java-generic-information.windup.xml:47

`                        <replacement>org.apache.camel.impl.{SupportClass}</replacement>`

Should be 

`                        <replacement>org.apache.camel.support.{SupportClass}</replacement>`

I realise this change wasn't made in this PR....

File: rules-reviewed/camel3/camel2/java-generic-information.windup.xml [hunk 1]
```diff
@@ -53,5 +53,370 @@
...
+                </javaclass>
+            </when>
+            <perform>
+                <iteration over="javaClass">
+                    <hint title="`org.apache.camel.impl.{moved}` has been moved." effort="1"
+                            category-id="mandatory">
+                        <message>The class `org.apache.camel.impl.{moved}` has been moved to `org.apache.camel.impl.engine` package.</message>
+                        <link title="Camel 3 - Migration Guide: Generic Information"
+                                href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_generic_information"/>
+                        <quickfix type="REPLACE" name="AggregationStrategy-replace">
```
Comment (m-brophy) [inline rules-reviewed/camel3/camel2/java-generic-information.windup.xml]:
```suggestion
                        <quickfix type="REPLACE" name="camel.impl-replace">
```

File: rules-reviewed/camel3/camel2/java-generic-information.windup.xml [hunk 2]
```diff
@@ -53,5 +53,370 @@
...
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="`org.apache.camel.util.toolbox.AggregationStrategies` has been moved" effort="1" category-id="mandatory">
+                    <message>
+                        The class `org.apache.camel.util.toolbox.AggregationStrategies` has been moved to `org.apache.camel.builder.AggregationStrategies`.
+                    </message>
+                    <link title="Camel 3 - Migration Guide: Generic Information"
+                            href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_generic_information"/>
+                    <quickfix type="REPLACE" name="AggregationStrategy-replace">
```
Comment (m-brophy) [inline rules-reviewed/camel3/camel2/java-generic-information.windup.xml]:
```suggestion
                    <quickfix type="REPLACE" name="AggregationStrategies-replace">
```

File: rules-reviewed/camel3/camel2/java-generic-information.windup.xml [hunk 3]
```diff
@@ -53,5 +53,370 @@
...
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="`org.apache.camel.management.JmxSystemPropertyKeys` has been moved" effort="1" category-id="mandatory">
+                    <message>
+                        The class `org.apache.camel.management.JmxSystemPropertyKeys` has been moved to ` org.apache.camel.api.management.JmxSystemPropertyKeys`.
+                    </message>
+                    <link title="Camel 3 - Migration Guide: Generic Information"
+                            href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_generic_information"/>
+                    <quickfix type="REPLACE" name="AggregationStrategy-replace">
```
Comment (m-brophy) [inline rules-reviewed/camel3/camel2/java-generic-information.windup.xml]:
```suggestion
                    <quickfix type="REPLACE" name="JmxSystemPropertyKeys-replace">
```

File: rules-reviewed/camel3/camel2/java-generic-information.windup.xml [hunk 4]
```diff
@@ -53,5 +53,370 @@
...
+            <javaclass references="org.apache.camel.language.Bean" as="annotations">
+                <location>IMPORT</location>
+           </javaclass>
+           </when>
+            <perform>
+                    <hint title="Annotation `org.apache.camel.language.Bean` method has been moved" effort="1" category-id="mandatory">
+                        <message>Annotation `org.apache.camel.language.Bean` has been moved to `org.apache.camel.language.bean.Bean`</message>
+                        <link href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_generic_information"
+                                title="Camel 3 - Migration Guide"/>
+                        <quickfix type="REPLACE" name="AggregationStrategy-replace">
```
Comment (m-brophy) [inline rules-reviewed/camel3/camel2/java-generic-information.windup.xml]:
```suggestion
                        <quickfix type="REPLACE" name="Bean-Annotation-replace">
```

File: rules-reviewed/camel3/camel2/java-generic-information.windup.xml [hunk 5]
```diff
@@ -53,5 +53,370 @@
...
+                <javaclass references="org.apache.camel.language.Simple" as="annotations">
+                    <location>IMPORT</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Annotation `org.apache.camel.language.Simple` method has been moved" effort="1" category-id="mandatory">
+                    <message>Annotation `org.apache.camel.language.Simple` has been moved to `org.apache.camel.language.simple.Simple`</message>
+                    <link href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_generic_information"
+                            title="Camel 3 - Migration Guide"/>
+                    <quickfix type="REPLACE" name="Annotation-replace">
```
Comment (m-brophy) [inline rules-reviewed/camel3/camel2/java-generic-information.windup.xml]:
```suggestion
                    <quickfix type="REPLACE" name="Simple-Annotation-replace">
```

File: rules-reviewed/camel3/camel2/java-generic-information.windup.xml [hunk 6]
```diff
@@ -53,5 +53,370 @@
...
+                <javaclass references="org.apache.camel.language.SpEL" as="annotations">
+                    <location>IMPORT</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Annotation `org.apache.camel.language.SpEL` method has been moved" effort="1" category-id="mandatory">
+                    <message>Annotation `org.apache.camel.language.SpEL` has been moved to `org.apache.camel.language.spel.SpEL`</message>
+                    <link href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_generic_information"
+                            title="Camel 3 - Migration Guide"/>
+                    <quickfix type="REPLACE" name="Annotation-replace">
```
Comment (m-brophy) [inline rules-reviewed/camel3/camel2/java-generic-information.windup.xml]:
```suggestion
                    <quickfix type="REPLACE" name="SpEL-Annotation-replace">
```

File: rules-reviewed/camel3/camel2/java-generic-information.windup.xml [hunk 7]
```diff
@@ -53,5 +53,370 @@
...
+                <javaclass references="org.apache.camel.{annotation}" as="annotations">
+                    <location>IMPORT</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Annotation `org.apache.camel.{annotation}` has been moved" effort="1" category-id="mandatory">
+                    <message>Annotation `org.apache.camel.{annotation}` has been moved to `org.apache.camel.spi.{annotation}`</message>
+                    <link href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_generic_information"
+                            title="Camel 3 - Migration Guide"/>
+                    <quickfix type="REPLACE" name="AAnnotation-replace">
```
Comment (m-brophy) [inline rules-reviewed/camel3/camel2/java-generic-information.windup.xml]:
```suggestion
                    <quickfix type="REPLACE" name="InvokeOnHeader-Annotation-replace">
```

File: rules-reviewed/camel3/camel2/java-generic-information.windup.xml [hunk 8]
```diff
@@ -53,5 +53,370 @@
...
+                <javaclass references="org.apache.camel.Constant" as="annotation">
+                    <location>IMPORT</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Annotation `org.apache.camel.Constant`has been removed" effort="1" category-id="mandatory">
+                    <message>Annotation `org.apache.camel.Constant` has been removed. Use `@Simple` instead.`</message>
+                    <link href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_generic_information"
+                            title="Camel 3 - Migration Guide"/>
+                    <quickfix type="REPLACE" name="Annotation-replace">
```
Comment (m-brophy) [inline rules-reviewed/camel3/camel2/java-generic-information.windup.xml]:
```suggestion
                    <quickfix type="REPLACE" name="Constant-Annotation-replace">
```

File: rules-reviewed/camel3/camel2/java-generic-information.windup.xml [hunk 9]
```diff
@@ -53,5 +53,370 @@
...
+                <javaclass references="org.apache.camel.builder.xml.XPathBuilder" as="class">
+                    <location>IMPORT</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Annotation `org.apache.camel.builder.xml.XPathBuilder`has been moved" effort="1" category-id="mandatory">
+                    <message>Class `org.apache.camel.builder.xml.XPathBuilder` has been moved to `org.apache.camel.language.xpath.XPathBuilder`.</message>
+                    <link href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_generic_information"
+                            title="Camel 3 - Migration Guide"/>
+                    <quickfix type="REPLACE" name="Annotation-replace">
```
Comment (m-brophy) [inline rules-reviewed/camel3/camel2/java-generic-information.windup.xml]:
```suggestion
                    <quickfix type="REPLACE" name="XPathBuilder-Annotation-replace">
```

File: rules-reviewed/camel3/camel2/java-generic-information.windup.xml [hunk 10]
```diff
@@ -53,5 +53,370 @@
...
+                <javaclass references="org.apache.camel.language.XPath" as="annotations">
+                    <location>IMPORT</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Annotation `org.apache.camel.language.XPath` has been moved" effort="1" category-id="mandatory">
+                    <message>Annotation `org.apache.camel.language.XPath` has been moved to `org.apache.camel.language.xpath.XPath`</message>
+                    <link href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_generic_information"
+                            title="Camel 3 - Migration Guide"/>
+                    <quickfix type="REPLACE" name="Annotation-replace">
```
Comment (m-brophy) [inline rules-reviewed/camel3/camel2/java-generic-information.windup.xml]:
```suggestion
                    <quickfix type="REPLACE" name="XPath-Annotation-replace">
```

File: rules-reviewed/camel3/camel2/java-generic-information.windup.xml [hunk 11]
```diff
@@ -53,5 +53,370 @@
...
+                <javaclass references="org.apache.camel.builder.xml.InvalidXPathExpression" as="annotations">
+                    <location>IMPORT</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Exception `org.apache.camel.builder.xml.InvalidXPathExpression` has been moved" effort="1" category-id="mandatory">
+                    <message>Exception `org.apache.camel.builder.xml.InvalidXPathExpression` has been moved to `org.apache.camel.language.xpath.InvalidXPathException `</message>
+                    <link href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_generic_information"
+                            title="Camel 3 - Migration Guide"/>
+                    <quickfix type="REPLACE" name="Annotation-replace">
```
Comment (m-brophy) [inline rules-reviewed/camel3/camel2/java-generic-information.windup.xml]:
```suggestion
                    <quickfix type="REPLACE" name="InvalidXPathExpression-Annotation-replace">
```

File: rules-reviewed/camel3/camel2/java-generic-information.windup.xml [hunk 12]
```diff
@@ -53,5 +53,370 @@
...
+                <javaclass references="org.apache.camel.processor.validation.PredicateValidationException" as="annotations">
+                    <location>IMPORT</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Exception `org.apache.camel.processor.validation.PredicateValidationException` has been moved" effort="1" category-id="mandatory">
+                    <message>Exception `org.apache.camel.processor.validation.PredicateValidationException` has been moved to `org.apache.camel.support.processor.validation.PredicateValidationException`</message>
+                    <link href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_generic_information"
+                            title="Camel 3 - Migration Guide"/>
+                    <quickfix type="REPLACE" name="Annotation-replace">
```
Comment (m-brophy) [inline rules-reviewed/camel3/camel2/java-generic-information.windup.xml]:
```suggestion
                    <quickfix type="REPLACE" name="PredicateValidationException-Annotation-replace">
```

File: rules-reviewed/camel3/camel2/xml-moved-components.windup.xml [hunk 13]
```diff
@@ -228,14 +228,13 @@
...
                     </xmlfile>
-
                 </or>
             </when>
             <perform>
                 <iteration over="dependencies-block">
                     <hint title="`camel-dataformat` component has been moved" effort="1" category-id="mandatory">
                         <message>`camel-dataformat` dependency required. Component has been moved from `camel-core` to separate artifact
-                            org.apache.camel:camel-dataformat that has to be added as a dependency to your project pom.xml file
+                            org.apache.camel:camel-dataformat that has to be added as a dependency to your projectom.xml file
```
Comment (m-brophy) [inline rules-reviewed/camel3/camel2/xml-moved-components.windup.xml]: This seems like a typo

**Pull request merged**